File: external/hypergraph-of-entity/src/main/java/armyant/hypergraphofentity/nodes/Node.java
package armyant.hypergraphofentity.nodes;

/**
 * Created by jldevezas on 2017-10-24.
 */
public abstract class Node {
    private String name;

    public Node() {

    }

    public Node(String name) {
        this.name = name;
    }

    public abstract String getType();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    @Override
    public String toString() {
        return "Node{" +
               "type='" + getType() + '\'' +
               "name='" + name + '\'' +
               '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Node node = (Node) o;

        return name.equals(node.name);
    }

    @Override
    public int hashCode() {
        return name.hashCode();
    }
}

File: external/cpp-impl/src/node/node.cpp
//
// Created by jldevezas on 3/6/18.
//

#include <node/node.h>

Node::Node() {
    Node::name = "";
}

Node::Node(std::string name) {
    Node::name = name;
}

const std::string &Node::getName() const {
    return name;
}

void Node::setName(const std::string &name) {
    Node::name = name;
}

bool Node::operator<(const Node &rhs) const {
    return name < rhs.name;
}

bool Node::operator>(const Node &rhs) const {
    return rhs < *this;
}

bool Node::operator<=(const Node &rhs) const {
    return !(rhs < *this);
}

bool Node::operator>=(const Node &rhs) const {
    return !(*this < rhs);
}
